thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobScheduler.java

```diff
@@ -309,13 +309,14 @@ public Long runBackfill(long functionId, DateTime backfillStartTime, DateTime ba
     }
 
     BackfillKey backfillKey = new BackfillKey(functionId, backfillStartTime, backfillEndTime);
-    Thread returnedThread = existingBackfillJobs.putIfAbsent(backfillKey, Thread.currentThread());
```

Comment:
I just realize existingBackfillJobs is a concurrency hashmap and this method could be triggered by multiple threads, so we need to keep the method putIfAbsent; otherwise, the code won't work as intended. 

For instance, if multiple threads arrive line 318 at the same time, then all of them passes the check at line 313 together. In the original design, only one of them may pass the check.

I suggest that we change the declare of existingBackfillJobs to ConcurrentHashMap type and don't change this part of code.